### PR TITLE
feat: add branch selection node [IDE-739]

### PR DESCRIPTION
**Pull request by acke**
### Description

- Preference state for Delta findings.
- Branch selection node to tree.
- Branch selection dialog.
- Handle FolderConfig notification.
- Support for doing a scan on a specific project.
- Enable/disable Net new issues option to the Viewmenu.
- Added handling of the case where Eclipse is opened without a project.

No open projects in Eclipse
image

Net new issues on View menu
image

Branch selection dialog
image

Scanning multiple projects with Delta scan
image

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/EclipsePreferenceState.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+
+import com.google.gson.Gson;
+
+import io.snyk.eclipse.plugin.utils.SnykLogger;
+import io.snyk.languageserver.protocolextension.messageObjects.FolderConfig;
+
+public class EclipsePreferenceState {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/EclipsePreferenceState.java:13]: Let's name it for what it does: it handles folder-specific settings. So something like `FolderPreferences` or `FolderConfigs` would probably be appropriate. `EclipsePreferenceState` is too abstract to be easily understandable.

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/EclipsePreferenceState.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	public List<String> getLocalBranches(String folderPath) {
+		String json = preferenceState.get(folderPath, null);
+		if (json != null) {
+			FolderConfig folderConfig = gson.fromJson(json, FolderConfig.class);
+			return folderConfig.getLocalBranches();
+		}
+		return List.of();
+	}
+
+	public void setBaseBranch(String folderPath, String newBaseBranch) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/EclipsePreferenceState.java:46]: This must be propagated to language server with a command.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/MenuHandler.java [hunk 3]
```diff
@@ -3,16 +3,36 @@
...
+		
+		if (firstElement instanceof JavaProject) {
+			project = ((JavaProject) firstElement).getProject();
+		}
+
+		if (firstElement instanceof IProject) {
+			project = (IProject) firstElement;
+		}
+
+		SnykExtendedLanguageClient.getInstance().triggerScan(project);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/MenuHandler.java:35]: Why is this changed to a project scan? previously, we had workspaceScan and folderScan - we should not have a breaking change here.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/RootNode.java [hunk 4]
```diff
@@ -23,6 +23,11 @@ public void reset() {
 		IProject[] allProjects = workspace.getRoot().getProjects();
 		List<IProject> openProjects = Arrays.stream(allProjects).filter(IProject::isOpen).collect(Collectors.toList());
 
+		if (openProjects.isEmpty()) {
+			var contentRoot = new ContentRootNode("No projects in workspace to scan", null);
+			this.addChild(contentRoot);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/RootNode.java:28]: shouldn't this exit early here?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 5]
```diff
@@ -180,6 +184,22 @@ public ProductTreeNode getProductNode(String product, String folderPath) {
...
+	@Override
+	public void resetContentRootNode(String project) {
+		if (project == null ) {
+			return;
+		}
+
+		for (TreeNode child : rootObject.getChildren()) {
+			if (child instanceof ContentRootNode) {
+				ContentRootNode contentRoot = (ContentRootNode) child;
+				if (project.startsWith(contentRoot.getName())) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:196]: we probably should use the path, not the name to identify the content root.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 6]
```diff
@@ -261,6 +281,75 @@ public void toggleIgnoresButtons() {
...
+	/*
+	 * Sets up for doing a Net New Issues scan.
+	 */
+	public void enableDelta() {
+		clearDeltaNodeListeners();
+
+		Display.getDefault().asyncExec(() -> {
+			if (this.treeViewer != null && !this.treeViewer.getTree().isDisposed()) {
+
+				TreeItem[] rootItems = getTreeViewer().getTree().getItems();
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:293]: why not use getRoot()?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 7]
```diff
@@ -261,6 +281,75 @@ public void toggleIgnoresButtons() {
...
+				for (TreeItem item : rootItems) {
+					ContentRootNode node = (ContentRootNode) item.getData();
+					String projectName = node.getName();
+					String projectPath = node.getPath().toString();
+					IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:298]: Probably better to use the path here :)

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 8]
```diff
@@ -261,6 +281,75 @@ public void toggleIgnoresButtons() {
...
+	/*
+	 * Disables Net New Issues scan, and starts a regular scan.
+	 */
+	public void disableDelta() {
+		clearDeltaNodeListeners();
+
+		SnykExtendedLanguageClient.getInstance().triggerScan(null);
+	}
+
+	private void clearDeltaNodeListeners() {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:332]: I'm not sure, I understand why we need a listener here.

File: plugin/src/main/java/io/snyk/languageserver/LsConfigurationUpdater.java [hunk 9]
```diff
@@ -57,10 +57,11 @@ Settings getCurrentSettings() {
         if (useTokenAuth) {
         	authMethod = "token";
         }
-        return new Settings(activateSnykOpenSource, activateSnykCodeSecurity, activateSnykCodeQuality, activateSnykIac, insecure, endpoint, additionalParams,
+		String enableDeltaFindings = preferences.getPref(Preferences.FILTER_DELTA_NEW_ISSUES, "");
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/LsConfigurationUpdater.java:60]: shouldn't the default be a "false"?

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 10]
```diff
@@ -391,6 +380,18 @@ public void snykScan(SnykScanParam param) {
 		setNodeState(param.getStatus(), affectedProductTreeNodes, issueCache);
 	}
 
+	@JsonNotification(value = LsNotificationID.SNYK_FOLDER_CONFIG)
+	public void folderConfig(FolderConfigsParam folderConfigParam) {
+		List<FolderConfig> folderConfigs = folderConfigParam != null ? folderConfigParam.getFolderConfigs() : List.of();
+		CompletableFuture.runAsync(() -> addAll(folderConfigs));
+	}
+
+	public void addAll(List<FolderConfig> folderConfigs) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:389]: This probably should be moved to a different file, maybe to the to be renamed PreferenceState thingy?

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 11]
```diff
@@ -528,16 +527,19 @@ public void publishDiagnostics316(PublishDiagnostics316Param param) {
...
-		var cacheHashMap = Collections.unmodifiableMap(issueCache.getCacheByDisplayProduct(productTreeNode.getProduct()));
+	private void populateFileAndIssueNodes(ProductTreeNode productTreeNode, String folderPath,
+			SnykIssueCache issueCache) {
+		var cacheHashMap = Collections
+				.unmodifiableMap(issueCache.getCacheByDisplayProduct(productTreeNode.getProduct()));
 		for (var kv : cacheHashMap.entrySet()) {
 			var fileName = kv.getKey();
 			var issues = new ArrayList<>(kv.getValue());
-			if(issues.isEmpty())
+			issues = filterIgnoredIssues(issues);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:541]: Why do we filter here? Shouldn't the filter just filter on the cache when accessing, not when inserting?

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/messageObjects/FolderConfig.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    public void setLocalBranches(List<String> localBranches) {
+        this.localBranches = localBranches != null ? localBranches : List.of();
+    }
+
+    public List<String> getAdditionalParameters() {
+        return additionalParameters;
+    }
+
+    public void setAdditionalParameters(List<String> additionalParameters) {
+        this.additionalParameters = additionalParameters != null ? additionalParameters : List.of();
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/messageObjects/FolderConfig.java:57]: This should be synchronized to project settings. How is it integrated with the PreferencePage?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java (hunk 8 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:332]: Yes, it's the listeners on the delta nodes. In SnykToolView we have a list of listeners ```itemListeners``` that we empty when the listeners are not needed anymore.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/MenuHandler.java (hunk 3 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/MenuHandler.java:35]: Confirming that both project (filepath) and workspace scans still works from the project explorer context menu.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java (hunk 6 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:277]: I need to TreeItems to update the text on them. Or is there another way to do this?

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java (hunk 7 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:282]: ResourcesPlugin.getWorkspace().getRoot().getProject(projectName) takes a name, I could not find a function that would return the project from the project name.

I could use IProject[] allProjects = workspace.getRoot().getProjects();
and loop through all the projects and compare with the path. 

for (IProject project : openProjects) {
Path path = ResourceUtils.getFullPath(project);
			...
			
			
Or do you have another way?

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+	public void addFolderConfig(FolderConfig folderConfig) {
+		preferenceState.put(folderConfig.getFolderPath(), gson.toJson(folderConfig));
+		try {
+			preferenceState.flush();
+		} catch (Exception e) {
+			SnykLogger.logError(e);
+		}
+	}
+
+	public List<String> getLocalBranches(String folderPath) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java:42]: Instead of strings, we could use `Path`

Comment (bastiandoetsch) [reply plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java:42]: Then we could use `Path.normalize()`

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+		String json = preferenceState.get(folderPath, null);
+		if (json != null) {
+			FolderConfig folderConfig = gson.fromJson(json, FolderConfig.class);
+			return folderConfig.getLocalBranches();
+		}
+		return List.of();
+	}
+
+	public void setBaseBranch(String folderPath, String newBaseBranch) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java:51]: Path maybe?

File: plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+	public void addAll(List<FolderConfig> folderConfigs) {
+		for (FolderConfig folderConfig : folderConfigs) {
+			addFolderConfig(folderConfig);
+		}
+	}
+
+	public FolderConfigsParam updateFolderConfigs() {
+		List<IProject> openProjects = ResourceUtils.getAccessibleTopLevelProjects();
+
+		List<String> projectPaths = openProjects.stream().map(project -> project.getLocation().toOSString())
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/properties/preferences/FolderConfigs.java:85]: you can use ResourceUtils.getFullPath() to get a Path object (I think).

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+		okButton.setLayoutData(new GridData(SWT.END, SWT.CENTER, false, false));
+		okButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				// Handle OK button press
+				String selectedBranch = dropdown.getText();
+				if (Arrays.asList(localBranches).contains(selectedBranch)) {
+					preferenceState.setBaseBranch(projectPath, selectedBranch);
+					shell.close();
+					SnykExtendedLanguageClient.getInstance().triggerScan(projectPath);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java:51]: This should not happen in the UI Thread. You could use for example: `CompletedFuture.runAsync(() -> {});`

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+		});
+
+		shell.pack();
+		shell.open();
+
+		while (!shell.isDisposed()) {
+			if (!display.readAndDispatch()) {
+				display.sleep();
+			}
+		}
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java:65]: lets's see if we really need it

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 18]
```diff
@@ -96,9 +99,21 @@ public void selectionChanged(SelectionChangedEvent event) {
 						LSPEclipseUtils.open(fileNode.getPath().toUri().toASCIIString(),
 								issueTreeNode.getIssue().getLSP4JRange());
 					}
+					boolean deltaEnabled = Preferences.getInstance()
+							.getBooleanPref(Preferences.FILTER_DELTA_NEW_ISSUES);
+					if (node instanceof ContentRootNode && deltaEnabled) {
+						ContentRootNode contentNode = (ContentRootNode) node;
+						String projectPath = contentNode.getPath().toString();
+						String[] localBranches = folderConfigs.getLocalBranches(projectPath).toArray(new String[0]);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:107]: here we would use the signature change above, passing the `Path` object to the folderConfigs

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 19]
```diff
@@ -96,9 +99,21 @@ public void selectionChanged(SelectionChangedEvent event) {
...
+					}
 				});
 			}
 		});
+
+		if (Preferences.getInstance().getBooleanPref(Preferences.FILTER_DELTA_NEW_ISSUES))
+			this.enableDelta();
 	}
 
 	private void registerTreeContextMeny(Composite parent) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:119]: Typo: should be `menu`

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 20]
```diff
@@ -118,6 +133,7 @@ public void setNodeText(BaseTreeNode node, String text) {
 		node.setText(text);
 		Display.getDefault().asyncExec(() -> {
 			this.treeViewer.refresh(node, true);
+			this.treeViewer.refresh(node.getParent(), true);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:136]: check if this is needed.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 21]
```diff
@@ -252,7 +268,46 @@ public void toggleIgnoresButtons() {
...
+	 */
+	public void enableDelta() {
+		if (this.treeViewer != null && !this.treeViewer.getTree().isDisposed()) {
+			BaseTreeNode[] children = (BaseTreeNode[]) getRoot().getChildren();
+
+			for (BaseTreeNode node : children) {
+				if (node instanceof ContentRootNode) {
+					ContentRootNode contentNode = (ContentRootNode) node;
+					String projectPath = contentNode.getPath().toString();
+					String projectName = ResourceUtils.getProjectByPath(contentNode.getPath()).getName();
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:284]: projectName should be contentRoot.getName()

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 22]
```diff
@@ -252,7 +268,46 @@ public void toggleIgnoresButtons() {
...
+			for (BaseTreeNode node : children) {
+				if (node instanceof ContentRootNode) {
+					ContentRootNode contentNode = (ContentRootNode) node;
+					String projectName = ResourceUtils.getProjectByPath(contentNode.getPath()).getName();
+					contentNode.setName(projectName);
+				}
+			}
+		}
 
+		SnykExtendedLanguageClient.getInstance().triggerScan(null);
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:310]: if you're in the UI thread, this should be called async, e.g. with CompletableFuture.runAsync...

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 23]
```diff
@@ -160,49 +170,23 @@ private void registerPluginInstalledEventTask() {
...
-
-	public LanguageServer getConnectedLanguageServer() {
-		if (this.ls != null) {
-			return ls;
-		}
-		return super.getLanguageServer();
-	}
-
-	public void triggerScan(IWorkbenchWindow window) {
+	public void triggerScan(String projectPath) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:173]: This could be a `Path` object.

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 24]
```diff
@@ -160,49 +170,23 @@ private void registerPluginInstalledEventTask() {
...
-						project = ((JavaProject) firstElement).getProject();
-					}
-
-					if (firstElement instanceof IProject) {
-						project = (IProject) firstElement;
-					}
 
-					if (project != null) {
-						runForProject(project.getName());
+				if (Preferences.getInstance().getBooleanPref(Preferences.FILTER_DELTA_NEW_ISSUES))
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:180]: Food for though: should we add a helper function to Preferences, `isDeltaEnabled()`?

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java [hunk 25]
```diff
@@ -378,6 +362,12 @@ public void snykScan(SnykScanParam param) {
 		setNodeState(param.getStatus(), affectedProductTreeNodes, issueCache);
 	}
 
+	@JsonNotification(value = LsNotificationID.SNYK_FOLDER_CONFIG)
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java:365]: We can merge LsCommandID with LsNotificationID and make it `LSContants`

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java (hunk 17 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java:65]: This loop effectively keeps the application responsive by continuously processing UI events. It handles user interactions, updates the GUI, and executes any asynchronous tasks submitted via Display.asyncExec(). The loop only exits when the shell is disposed, typically when the user closes the application window.

File: plugin/src/main/java/io/snyk/languageserver/protocolextension/messageObjects/FolderConfig.java (hunk 12 above)
Comment (acke) [reply plugin/src/main/java/io/snyk/languageserver/protocolextension/messageObjects/FolderConfig.java:57]: WIll be done in a separate PR.

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+import io.snyk.eclipse.plugin.utils.SnykLogger;
+import io.snyk.languageserver.protocolextension.SnykExtendedLanguageClient;
+
+public class BaseBranchDialog {
+	private FolderConfigs preferenceState = FolderConfigs.getInstance();
+
+	public BaseBranchDialog() {
+	}
+
+	public void baseBranchDialog(Display display, Path projectPath, String[] localBranches) {
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java:29]: please rename as discussed

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java [hunk 27]
```diff
@@ -0,0 +1,71 @@
...
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Shell;
+
+import io.snyk.eclipse.plugin.properties.preferences.FolderConfigs;
+import io.snyk.eclipse.plugin.utils.SnykLogger;
+import io.snyk.languageserver.protocolextension.SnykExtendedLanguageClient;
+
+public class BaseBranchDialog {
+	private FolderConfigs preferenceState = FolderConfigs.getInstance();
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/BaseBranchDialog.java:24]: please rename as discussed

File: plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java [hunk 28]
```diff
@@ -93,9 +98,22 @@ public void selectionChanged(SelectionChangedEvent event) {
 						LSPEclipseUtils.open(fileNode.getPath().toUri().toASCIIString(),
 								issueTreeNode.getIssue().getLSP4JRange());
 					}
+					boolean deltaEnabled = Preferences.getInstance()
```
Comment (bastiandoetsch) [inline plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java:101]: let's use the helper function you added in Preferences

**Review by bastiandoetsch — APPROVED** [review]
